pkgs/tools/misc/dua/default.nix

```diff
@@ -3,32 +3,37 @@
 , fetchFromGitHub
 , stdenv
 , darwin
+, nix-update-script
 }:
 
 rustPlatform.buildRustPackage rec {
   pname = "dua";
-  version = "2.29.0";
+  version = "2.29.1";
 
   src = fetchFromGitHub {
     owner = "Byron";
     repo = "dua-cli";
     rev = "v${version}";
-    hash = "sha256-rO9k1/HOwVJF/QCT2sZy4L0Mv26CiUj9Zafliffj68A=";
+    hash = "sha256-YOXK6u9KF2FnICMFEj5IEZG2OQ6PaKQsShp7/gbSK4Y=";
     # Remove unicode file names which leads to different checksums on HFS+
     # vs. other filesystems because of unicode normalisation.
     postFetch = ''
       rm -r $out/tests/fixtures
     '';
   };
 
-  cargoHash = "sha256-qn1QDiYHcygomOFwFEy00wsMykrQ9/84Ed4nAUTlA1k=";
+  cargoPatches = [ ./add-cargo-lock.patch ];
+
+  cargoHash = "sha256-Km8I5t6xDaCYn4Z8v7LLDtX4sk89iqnliECPiV7Ed5o=";
 
   buildInputs = lib.optionals stdenv.isDarwin [
     darwin.apple_sdk.frameworks.Foundation
   ];
 
   doCheck = false;
 
+  passthru.updateScript = nix-update-script { };
```

Comment:
I don't think this will work, because the patch will always need to be updated.

Reply:
Gotcha, removed it